cmd/gnoblog-cli/util.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/adrg/frontmatter"
+)
+
+func parsePost(reader io.Reader) (*post, error) {
+	var p post
+	rest, err := frontmatter.MustParse(reader, &p)
+	if err != nil {
+		return nil, fmt.Errorf("invalid post frontmatter: %w", err)
+	}
+
+	body := string(rest)
+	p.Title, err = extractTitle(body)
+
+	p.Body = removeTitle(body, p.Title)
+	p.Tags = removeWhitespace(p.Tags)
+	if err != nil {
+		return nil, err
+	}
+
+	if p.PublicationDate == nil {
+		now := time.Now()
+		p.PublicationDate = &now
+	}
+
+	return &p, nil
+}
+
+// extractTitle extracts H1 (Title) from the body of the post
+func extractTitle(body string) (string, error) {
+	scanner := bufio.NewScanner(strings.NewReader(body))
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "# ") {
+			return line[2:], nil
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return "", err
+	}
+
+	return "", fmt.Errorf("no H1 header found")
+}
+
+// removeTitle removes the first occurrence of the Title from the body of the post
+func removeTitle(body, title string) string {
+	t := "# " + title + "\n"
+	return strings.Replace(body, t, "", 1)
+}
+
+// removeWhitespace loops over a slice of tags (strings) and truncates each tag
+// by removing whitespace
+// ie, []string{"example spaced tag"} -> []string{"examplespacedtag"}
+func removeWhitespace(tags []string) []string {
+	for i, tag := range tags {
+		tags[i] = strings.Replace(tag, " ", "", -1)
+	}
+
+	return tags
+}
```

Comment:
either return a new slice or return nothing: to avoid confusion and unexpected outcomes for the caller

Reply:
I made a check to never run this function is the tags slice is empty, and this now returns a newly allocated slice: [eb04b51](https://github.com/gnolang/blog/pull/63/commits/eb04b51f96c373179953efbf4cb2e1826fb52581)